### ISSUE TITLE
Docker service on a node can reload config only twice

### ISSUE THREAD

**Issue by mmorhun** [codenvy/codenvy#1211]
When try to reload config of docker (without restart) on a Codenvy node more then two times changes are ignored.

**Reproduction Steps:**
1. Run `docker info` and remember existing labels (if no labels configured section is not appeared)
2. Edit / create file `/etc/docker/daemon.json` and add a new label. Example of content of created file: 
  `{"labels":["com.codenvy.some-label=somevalue"]}`
3. Reload docker config. Do NOT restart docker, use sighup signal:
  `kill -SIGHUP $(<"/var/run/docker.pid")`
  Use `docker info` to ensure, that the label is present.
4. Change label value and reload config again. Check again, that changes are applied.
5. Change label value again and again reload config (3rd time).

**Expected behavior:**
Config should be reloaded again and next times.

**Observed behavior:**
Nothing changes. Config is left as after second reloading.
To apply new changes docker restart is needed.

**Codenvy version:** 5.0.0-M8-SNAPSHOT
**OS and version:** CentOS Linux 7 (Core) 3.10.0-327.10.1.el7.x86_64 
**Docker version:** Version: 1.11.1  API version: 1.23
**Swarm:** Server Version: swarm/1.2.3

**Additional information:**

Problem only started happening recently, didn't happen in an older version of Codenvy: Yes
Problem can be reliably reproduced: Yes

**Comment by mmorhun** [comment]
After updating to docker 1.13.0 and swarm 1.2.6 behaviour changes.
After SIGHUP signal with empty json config (`{}`) docker answers with:
```

**Comment by garagatyi** [comment]
AFAIK docker 1.13 is still RC. PLEASE create an issue in docker repo. Maybe they will fix it before 1.13 GA

**Comment by mmorhun** [comment]
I didn't know, that docker 1.13 is still RC, but I was about to create the issue.
So, I've created [issue](https://github.com/docker/docker/issues/30680) in docker repo.

**Comment by TylerJewell** [comment]
1.13 is ga

**Comment by garagatyi** [comment]
My bad, sorry. It was bugfix RC.

**Comment by mmorhun** [comment]
I've reproduced it without Codenvy. Steps:
1) Run Zookeeper: `docker run -p 0.0.0.0:2181:2181 -d zookeeper:3.4.9`
2) Create 2 nodes using docker machine: `docker-machine create -d virtualbox --virtualbox-memory 4096 --virtualbox-cpu-count=2 --engine-opt="cluster-store=zk://<host-ip>:2181" --engine-opt="cluster-advertise=eth1:2375" --engine-opt="host=tcp://0.0.0.0:2375" --engine-env="DOCKER_TLS=no" node1`
3) Run Swarm: `docker run -d -v /tmp/node_list:/node_list -p 127.0.0.1:23751:2375 swarm:1.2.6 manage -H 0.0.0.0:2375 file:///node_list`, where `/tmp/node_list` is text file:
```
192.168.99.100:2376
192.168.99.101:2376
```
(ip addresses could be different, see `docker-machine ls` for correct values)

**Comment by mmorhun** [comment]
This problem occurs when docker on a node is started with `cluster-store` and `cluster-advertise` options. Otherwise everything works fine.
In case, when only `cluster-store` is set, docker hungs up on 3rd config reload. Logs:
- after 1st SIGHUP:
```
ime="2017-02-07T13:19:18.661579317Z" level=info msg="Got signal to reload configuration, reloading from: /etc/docker/daemon.json" 
time="2017-02-07T13:19:18.661771835Z" level=debug msg="Initializing discovery service" name=zk uri="<host-ip>:2181" 
time="2017-02-07T13:19:18.661784409Z" level=info msg="Initializing discovery without TLS" 
time="2017-02-07T13:19:18.661811447Z" level=debug msg="Option Experimental: false" 
time="2017-02-07T13:19:18.661817781Z" level=debug msg="Option DefaultDriver: bridge" 
time="2017-02-07T13:19:18.661822227Z" level=debug msg="Option DefaultNetwork: bridge" 
time="2017-02-07T13:19:18.661826658Z" level=debug msg="Option OptionKVProvider: zk" 
time="2017-02-07T13:19:18.661831655Z" level=debug msg="Option OptionKVProviderURL: <host-ip>:2181" 
time="2017-02-07T13:19:18.661840320Z" level=debug msg="Reset Max Concurrent Downloads: 4" 
time="2017-02-07T13:19:18.661845701Z" level=debug msg="Reset Max Concurrent Uploads: 5" 
```
2nd:
```
time="2017-02-07T13:20:18.620647092Z" level=info msg="Got signal to reload configuration, reloading from: /etc/docker/daemon.json" 
time="2017-02-07T13:20:18.620767361Z" level=error msg="Error reconfiguring the daemon: discovery service not supported"
```
3rd:
```
time="2017-02-07T13:20:35.780443737Z" level=info msg="Got signal to reload configuration, reloading from: /etc/docker/daemon.json"
```

**Comment by mmorhun** [comment]
Issue in docker is resolved, fix is merged for 1.14.0 milestone.
